### PR TITLE
feat: [OSM-516] Begin replacing `depTree` and prepare for `runtimeAssemblies` parsing

### PR DESCRIPTION
**Pull request by dotkas**
# Something like 11.100 lines are fixtures, I swear! 🤞🏻 🥵 

## Background

As a part of the larger initiative with [OSM-516] we are introducing an `option` in the plugin `dotnet-runtime-resolution` (any ideas for a better name? 😬 ) in order to separate new functionality with existing, not break backwards compatibility and keep things isolated while we transition.

How this option will be populated from the CLI is still not completely clear, but I am debating it with team Hammerhead. Currently considering either 

* A command line flag, that we then will have to keep forever due to the nature of versions and compatibility of the CLI, or
* Organization feature flags getting picked up by the CLI, which it already supports

## What does this PR do?

Adds `depGraph` generation instead of the deprecated and more cumbersome depTrees, currently only for `netX.X` target frameworks, to keep things tight. As a side effect solves the issue with `freqSystemDependencies` ([SUP-564](https://snyksec.atlassian.net/browse/SUP-564)). We already have a working PoC in the isolated Dotnet solution, which we now want to port to this plugin instead of the built-in solution. 

The isolated builds will keep doing dll disassembly, while this will traverse the assembly versions in the generated `.json` file manifests.

This PR is mostly a copy-paste of the existing solution, molded a bit to fit here.

[OSM-516]: https://snyksec.atlassian.net/browse/OSM-516?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Random benchmark

Another nice benefit for fixture sizes:

DepGraphs:
```
 79K Jul 17 08:32 expected_depgraph.json
```
DepTrees:
```
 11M May  3 23:15 expected.json
```

[SUP-564]: https://snyksec.atlassian.net/browse/SUP-564?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/csproj.spec.ts
Comment (dotkas) [inline test/csproj.spec.ts:1]: This is unrelated, just refactored it a bit. Was too lazy to make a PR just for that.

File: test/parse-dotnet-cli.spec.ts [hunk 1]
```diff
@@ -17,39 +17,29 @@ describe('when parsing .NET CLI', () => {
...
-      );
-      const result = await plugin.inspect(projectPath, manifestFile);
-      expect(result).toEqual(expectedTree);
-    });
-
-    it('parse dotnet-cli project and traverse packages', async () => {
-      const projectPath = './test/stubs/dotnet_p_g';
-      const manifestFile = 'obj/project.assets.json';
-      const expectedTree = JSON.parse(
-        fs.readFileSync('./test/stubs/dotnet_p_g/expected.json', 'utf-8'),
```
Comment (dotkas) [inline test/parse-dotnet-cli.spec.ts:45]: Deleted this test, as it was pointless. It's supposed to traverse `_packages` but the folder doesn't exist. Another test is already doing it.

File: lib/nuget-parser/dotnet-core-v2-parser.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  name: string;
+  version: string;
+}
+
+export interface Assembly {
+  name: string;
+  version: string;
+}
+
+export type AssemblyVersions = Record<string, string>;
```
Comment (dotkas) [inline lib/nuget-parser/dotnet-core-v2-parser.ts:27]: Currently not used, next PR will start to.

File: lib/nuget-parser/dotnet-core-v2-parser.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+// Dependencies that starts with these are discarded
+const FILTERED_DEPENDENCY_PREFIX = ['runtime'];
```
Comment (dotkas) [inline lib/nuget-parser/dotnet-core-v2-parser.ts:30]: These pollute the generated dependency graph, and are (to the best of my understanding) not relevant until we start allowing users to specify `TargetRuntimeIdentifier` in their projects.

File: test/parse-dotnet-v2.spec.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import * as path from 'path';
+import * as plugin from '../lib';
+
+describe('when generating depGraphs and runtime assemblies using the v2 parser', () => {
+  it.each([
+    {
+      description: 'parse dotnet 6.0',
+      projectPath: './test/stubs/dotnet_6',
+    },
+  ])(
```
Comment (dotkas) [inline test/parse-dotnet-v2.spec.ts:12]: This will be expanded greatly in the next PR, so made it table-driven already.

File: lib/index.ts [hunk 5]
```diff
@@ -62,6 +66,33 @@ export async function inspect(root, targetFile, options?) {
...
+    const result = await nugetParser.buildDepGraphFromFiles(
+      root,
+      targetFile,
+      manifestType,
+      options['assets-project-name'],
+      options['project-name-prefix'],
+    );
+    return {
+      dependencyGraph: result.dependencyGraph,
+      package: 'n/a', // TODO: Will remove when everything is ported to depGraphs
```
Comment (dotkas) [inline lib/index.ts:87]: If I made it optional this PR would be 500 lines longer with type guards.

File: test/csproj.spec.ts
Comment (dotkas) [inline test/csproj.spec.ts:1]: Ignore this file, just refactor, and was too lazy to make a separate PR.

**Review by DOlufemi — APPROVED** [review]
LGTM

**Pull request merged**